snapcraft/plugins/roscore.py

```diff
@@ -81,3 +84,35 @@ def snap_fileset(self):
             '-' + os.path.join(rospath, 'setup.sh'),
             '-' + os.path.join(rospath, '_setup_util.py')
         ])
+
+    def _finish_build(self):
+        rospath = os.path.join(self.installdir, 'opt', 'ros',
+                               self.options.rosdistro)
+
+        # Fix all shebangs to use the in-snap python.
+        pattern = re.compile(r'#!.*python')
+        for root, directories, files in os.walk(rospath):
+            for fileName in files:
+                with open(os.path.join(root, fileName), 'r+') as f:
+                    try:
+                        original = f.read()
+                    except UnicodeDecodeError:
+                        # This was probably a binary file. Skip it.
+                        continue
+
+                    replaced = pattern.sub(r'#!/usr/bin/env python', original)
+                    if replaced != original:
+                        f.seek(0)
+                        f.truncate()
+                        f.write(replaced)
+
+        # Also replace the python usage in 10.ros.sh to use the in-snap python.
+        setup_util_file = os.path.join(rospath,
+                                       'etc/catkin/profile.d/10.ros.sh')
+        if os.path.isfile(setup_util_file):
+            with open(setup_util_file, 'r+') as f:
+                pattern = re.compile(r'/usr/bin/python')
+                replaced = pattern.sub(r'python', f.read())
+                f.seek(0)
+                f.truncate()
+                f.write(replaced)
```

Comment:
I think this is clearly composed of two blocks, so I would extract two methods. One like _fix_shebangs or _patch_shebangs. The other something like _use_snap_python.
If you find really nice names for them, maybe you can get rid of the call to _finish_build (which has a name that doesn't say much) by directly calling these methods.
I like small methods with nice names. Lots of them. Sometimes I go too far, so feel free to disagree :)